Update NFTS Pallet Page (#4486)
w3f/polkadot-wiki: docs/learn/learn-nft-pallets.md

The pallet comes with a new way to configure NFTs, as well as configure collections and items.
Pallet-level [feature flags](https://github.com/paritytech/substrate/pull/12367) allow disabling
functionalities not needed in the runtime.

### Roles

Setting up a collection implies different roles with different permissions:

- Owner:

  - destroy collection (to destroy the collection, there should be 0 items left).
  - redeposit: re-evaluate the deposit on some items.
  - set team: change the collection’s Issuer, Admin, Freezer.
  - set collection max supply: set the maximum number of items for a collection.
  - lock collection: this can include making a collection’s items non-transferable, fixing its max
    supply, and locking collection metadata and attributes.

- Admin:

>   - set attribute and metadata of a collection.

Reviewer: "set attribute**s**"

>   - set attributes pre-signed: set attributes for an item by providing the pre-signed approval.

Reviewer: "set attributes for an item by providing the pre-signed approval." -> "set attributes for an item by providing the Admin pre-signed approval."

  - lock item properties: lock item metadata and attributes.

:::info

> Note that an Admin account cannot burn or transfer items it does not own.

Reviewer: I don't think we need to say this

:::

- Freezer:

>   - lock item transfer: lock the possibility of transferring an item for all holders.

Reviewer 2: Rephrased it. I hope I did not change the meaning of what was intended.

```suggestion
  - lock item transfer: lock all holders from the possibility of transferring an item.
```

>   - lock item transfer: lock all holders from the possibility of transferring an item.

Reviewer: "lock all holders from the possibility of transferring an item." -> "disallow further item transfers."

>   - unlock item transfer: lift a previous lock to transfer an item for all holders.

Reviewer: "lift a previous lock to transfer an item for all holders." -> "lift a previous lock to transfer an item."
It's better to avoid saying "all holders" since there is only one owner of the item

- Issuer
  - mint

>   - force mint (with custom item configuration): mint an item of a particular collection from a

Reviewer: I would remove the `: mint an item of a particular collection from a privileged origin.`

    privileged origin.

>   - mint pre-signed: mint an item by providing the pre-signed approval.

Reviewer: "mint an item by providing the pre-signed approval." -> "mint an item by providing the Issuer pre-signed approval."

  - update mint settings.

For simple collections, the same account has all the roles by default. Still, for complex
collections separate accounts can take each role with their responsibilities (e.g. items issuance).
The key can be rotated for those roles. The owner's account is used to setup the collection, and its

> private key is kept in cold storage.

Reviewer: I would remove that

Reviewer 3: yeah, owner can keep private key on a paper stamped on the fridge, we can't control that

> Those roles can also be set to `none` without the ability to change them back. This is useful when a

Reviewer: "Any role (Issuer/Admin/Freezer) can be set to `None` without the ability to change it back. This is useful when a"

> collection is created, and all the items are minted without the possibility of minting any more

Reviewer: "This is useful when a collection is created, and all the items are minted. Now, by setting roles to `None` we remove the possibility of minting any more items, changing the metadata, or disallowing some item's transfer."

Reviewer 3: comma before or isn't necessary

:::info

 The user can decide to lock an item or collection’s metadata, attributes, and settings. Also, a locking mechanism can prevent unauthorized and unprivileged transfers (unprivileged

> actions can be re-allowed anytime).

Reviewer 3: this paragraph seems out of place, the functionality should be described in the roles section

:::info

An item or collection’s metadata, attributes, and settings can be locked. The user can decide what

> to lock. Also, a locking mechanism can prevent unauthorized and unprivileged transfers (unprivileged

Reviewer 2: ```suggestion
 The user can decide to lock an item or collection’s metadata, attributes, and settings. Also, a locking mechanism can prevent unauthorized and unprivileged transfers (unprivileged
```

account owning the NFT and the collection ID. You can also see all your collections by selecting the
`collectionAccount` extrinsic.

When a new collection is created, a new ID will be generated and assigned to it. When a collection

> is destroyed, and no one can pick up the collection ID again (including the owner).

Reviewer: "When a collection is destroyed, then no one can pick up that ID again (including the owner)."

Reviewer 3: The original text was correct.
"When a collection is destroyed, no one can pick up that ID again (including the owner)."

```

- Upload the metadata file to Pinata and get the updated CID.
- After minting your NFT on the Polkadot-JS UI, you can add the CID. Go to Developer > Extrinsics
  and select the `nfts.setCollectionMetadata` (for collections) or `nfts.setMetadata` (for single
  NFTs) extrinsic. Under the `data: Bytes` field you can enter the CID or upload the metadata file.

The collection can be created and its item minted before uploading the NFT file and related
metadata. The minting process on-chain will assign a collection and item ID to your account. Those
IDs will later be populated with NFT files, metadata, and attributes. Once you upload the NFT files

> and related data, the abovementioned extrinsics can be used to update a collection or item.

Reviewer 2: ```suggestion
and related data, the above-mentioned extrinsics can be used to update a collection or item.
```
